OrbitLinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -16,13 +16,16 @@ void UprobesUnwindingVisitor::visit(SamplePerfEvent* event) {
   const std::vector<unwindstack::FrameData>& full_callstack =
       unwinder_.Unwind(current_maps_.get(), event->GetRegisters(),
                        event->GetStackData(), event->GetStackSize());
-  if (!full_callstack.empty()) {
+  // Some samples can actually fall inside u(ret)probes code. Discard them,
+  // because when they are unwound successfully the result is wrong.
+  if (!full_callstack.empty() &&
+      full_callstack.front().map_name != "[uprobes]") {
     Callstack returned_callstack{
         event->GetTid(),
         CallstackFramesFromLibunwindstackFrames(full_callstack),
         event->GetTimestamp()};
     listener_->OnCallstack(returned_callstack);
-  } else if (unwind_error_counter_ != nullptr) {
+  } else if (full_callstack.empty() && unwind_error_counter_ != nullptr) {
```

Comment:
Is this the only please where unwind_error_counter_ is incremented?

maybe add IncrementUnwindErrorCounter() method which does the check and increments if not null.. 

in which case

if (full_callstack.empty()) {
  Increment...
  return;
}

if (full_callstack.front().map_name != "[uprobes]") {
  ...
}

Reply:
Yes, this is the only place. I changed the code a bit as I added another counter for discarded samples so now the code is much closer to your suggestion even if I haven't introduced a new method.